aip/client-libraries/4236.md

```diff
@@ -1,53 +1,132 @@
 ---
 id: 4236
-state: reviewing
+state: approved
 created: 2024-05-16
 ---
 
 # Version-aware clients
 
-APIs can annotate services with [`google.api.api_version`][]. If
-`google.api.api_version` is specified, version-aware clients **must**
-include the value of `google.api.api_version` in the request to the API.
+APIs can annotate [API interfaces][interface def] with
+[`google.api.api_version`][annotation]. If `google.api.api_version` is
+specified, version-aware clients **must** include the value of
+`google.api.api_version` in the request to the API and in the documentation of
+the per-API interface clients.
 
-### Expected Generator and Client Library Behavior
+## Generator and client library behavior
 
-If a service is annotated with `google.api.api_version`, client library
-generators **must** include either an HTTP query parameter `$apiVersion`
+If an API interface is annotated with `google.api.api_version`, client
+library generators **must** include either an HTTP query parameter `$apiVersion`
```

Comment:
Do we want to say anything about specifying the version over gRPC transport? I know our canary use case is REST, but if we have plan for the gRPC implementation we should state it. Or maybe state that info on the gRPC side of things is a WIP?

Reply:
Good shout, I'll add reference to gRPC  metadata in the header phrase.